### PR TITLE
[add] Save filters to dashboard

### PR DESCRIPTION
**Pull request by roganw**
Save the filters of filter box to current dashboard.

### CODE REVIEW

File: superset/models/core.py [hunk 1]
```diff
@@ -322,6 +323,18 @@ def table_names(self):
...
     def url(self):
+        if self.json_metadata:
+            # add default_filters to the preselect_filters of dashboard
+            json_metadata = json.loads(self.json_metadata)
+            default_filters = json_metadata.get('default_filters')
+            if default_filters:
+                filters = ''
+                if isinstance(default_filters, str):
+                    filters = urllib.quote(default_filters)
+                elif isinstance(default_filters, unicode):
```
Comment (mistercrunch) [inline superset/models/core.py]: is this py3 compatible? can we add test coverage?

File: superset/assets/javascripts/dashboard/Dashboard.jsx [hunk 2]
```diff
@@ -195,6 +199,13 @@ export function dashboardContainer(dashboard, datasources, userid) {
       return f;
     },
     addFilter(sliceId, col, vals, merge = true, refresh = true) {
+      // If slice doesn't exist, remove the related parameters in preselect_filters.
```
Comment (mistercrunch) [inline superset/assets/javascripts/dashboard/Dashboard.jsx]: this function calls `this.updateFilterParamsInUrl` 3 times. Perhaps clearer logic would be an if statement that validates that there is a slice and column, put all the logic within that block, and call `this.updateFilterParamsInUrl` once at the very end

**Review by mistercrunch — CHANGES_REQUESTED** [review]
see comments

File: superset/models/core.py (hunk 1 above)
Comment (RoganW) [inline superset/models/core.py]: Thanks, I will check these later.

**Comment by RoganW** [comment]
@mistercrunch Thanks, code optimized.

**Review by mistercrunch — APPROVED** [review]

**Comment by mistercrunch** [comment]
This is great work, thanks for this feature!

**Pull request merged**